controllers/external-accounts.js

```diff
@@ -71,14 +71,20 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
-    try {
-      const unverifiedRoleId = config.get("discordUnverifiedRoleId");
-      await discordServices.removeRoleFromUser(unverifiedRoleId, attributes.discordId, req.userData);
-    } catch (error) {
-      logger.error(`Error getting external account data: ${error}`);
-      return res.boom.internal("Role Deletion failed. Please contact admin.", {
-        message: "Role Deletion failed. Please contact admin.",
-      });
+    const unverifiedRoleId = config.get("discordUnverifiedRoleId");
+    const unverifiedRoleRemovalResponse = await removeDiscordRoleUtils.removeDiscordRoleFromUser(
+      req.userData,
+      attributes.discordId,
+      unverifiedRoleId
+    );
+
+    if (!unverifiedRoleRemovalResponse.success) {
+      return res.boom.internal(
+        `User details updated but ${unverifiedRoleRemovalResponse.message}. Please contact admin`,
+        {
+          message: `User details updated but ${unverifiedRoleRemovalResponse.message}. Please contact admin`,
+        }
+      );
     }
 
     return res.status(204).json({ message: "Your discord profile has been linked successfully" });
```

Comment:
For 204 status, why are you sending body?

Reply:
I've not touched that part of the code, it was already there. Also if I remove the body, I don't know how the frontend will behave. So it's beyond my scope.